text/0000-add-logical-operators.md

```diff
@@ -0,0 +1,97 @@
+and by short-circuiting we can avoid computing them at all sometimes.
+
+
+## Detailed design
+
+Add `{{and}}`, `{{or}}` and `{{not}}` helpers.
+
+#### `{{and}}`
+Takes at least two positional arguments. Raises an error if invoked with less than two arguments.
+It evaluates arguments left to right, returning the first one that is not _truthy_ (**by handlebar's definition of truthiness**)
+or the right-most arguments if **all** evaluate to _truthy_.
+This is *NOT* equivalent to the `{{and}}` helper from `ember-truth-helpers` because unlike this proposed helper, the one in `ember-truth-helpers`
+uses Javascript's definition of truthiness.
+
+#### `{{or}}`
+Takes at least two positional arguments. Raises an error if invoked with less than two arguments.
+It evaluates arguments left to right, returning the first one that is _truthy_ (**by handlebar's definition of truthiness**) or the
+right-most argument if all evaluate to _falsy_.
+This is *NOT* equivalent to the `{{or}}` helper from `ember-truth-helpers` because unlike this proposed helper, the one in `ember-truth-helpers`
+uses Javascript's definition of truthiness.
+
+#### `{{not}}`
+Unary operator. Raises an error if invoked with more than one positional argument. If the given value evaluates to a _truthy_ value (**by handlebar's definition of truthiness**),
+the `false` is returned. If the given value evaluates to a _falsy_ value (**by handlebar's definition of truthiness**) then it returns `true`.
+This is *NOT* equivalent to the `{{not}}` helper from `ember-truth-helpers` because unlike this proposed helper, the one in `ember-truth-helpers`
+uses Javascript's definition of truthiness.
+
+#### Handlebar's definition of truthiness
+This is the most important detail of this proposal because it's where it deviates from `ember-truth-helpers`.
+Handlebars has its own definition of _truthyness_, which is similar to Javascripts except that empty arrays are
+considered **falsy**, while in JS are considered **truthy**.
+
+
+This RFC intentionally leaves the implementation details unspecified, but one can think of those helpers as macros that
+expand to combinations of `if`s.
+
+##### `{{and}}`
+- `{{and a b}}` is equivalent to `{{if a b a}}`
+- `{{and a b c}}` is equivalent to  `{{if a (if b c b) a}}`
+- and so on
+
+##### `{{or}}`
+- `{{or a b}}` is equivalent to `{{if a a b}}`
+- `{{and a b c}}` is equivalent to  `{{if a a (if b b c)}}`
+- and so on
+
+##### `{{not}}`
+- `{{not a}}` is equivalent to `{{if a false true}}`
+
+## How we teach this
+
+The introduction of these helpers does not impact the current mental model for Ember applications.
+
+In addition to API and Guides documentation with illustrative examples of usage of the various helpers, stressing out
+where the definition of truthiness of handlebars is different from the one in Javascript.
```

Comment:
I think we need to beef this section a bit more. Specifically, I _do_ think this changes some of the mental models. For example, commonly we punt folks into JS-land instead of using ember-truth-helpers in the guides/tutorial (e.g. we might suggest using `computed.not`). This would definitely change if we introduce these helpers.